src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs

```diff
@@ -32,19 +34,20 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
             return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
         }
 
-        Task CheckWithReceiver()
+        async Task CheckWithReceiver()
         {
-            var client = ClientConnections.GetOrAdd(ConnectionKey, _ => CreateClient());
-            var receiver = ServiceBusReceivers.GetOrAdd(
-                $"{nameof(AzureServiceBusQueueHealthCheck)}_{ConnectionKey}",
-                client.CreateReceiver(Options.QueueName));
+            var client = await ClientCache.GetOrAddAsyncDisposableAsync(ConnectionKey, _ => CreateClient()).ConfigureAwait(false);
+            var receiver = await ClientCache.GetOrAddAsyncDisposableAsync(
+                $"{nameof(AzureServiceBusQueueHealthCheck)}_{queueKey}",

```

Comment:
Thit key is computed every time. Please move `{nameof(AzureServiceBusQueueHealthCheck)}` into `queueKey` and make it field.